rero_ils/modules/items/serializers/csv.py

```diff
@@ -72,141 +46,34 @@ def serialize_search(self, pid_fetcher, search_result, links=None,
-                    .source(['pid', 'item_pid.value', 'start_date',
-                            'end_date', 'state', '_created'])
-                agg = A('terms', field='item_pid.value', size=chunk_size)
-                loan_search.aggs.bucket('loans_count', agg)
-
-                loan_search = loan_search.extra(
-                    collapse={
-                        'field': 'item_pid.value',
-                        "inner_hits": {
-                            "name": "most_recent",
-                            "size": 1,
-                            "sort": [{"_created": "desc"}],
-                        }
-                    }
-                )
-                # default results size for the execute method is 10.
-                # We need to set this to the chunk size"
-                results = loan_search[0:chunk_size].execute()
-                agg_buckets = {}
-                for result in results.aggregations.loans_count.buckets:
-                    agg_buckets[result.key] = result.doc_count
-                loans = {}
-                for loan_hit in results:
-                    # get most recent loans
-                    loan_data = loan_hit.meta.inner_hits.most_recent[0]\
-                        .to_dict()
-                    item_pid = loan_data['item_pid']['value']
-                    loans[item_pid] = {
-                        'loans_count': agg_buckets.get(item_pid, 0),
-                        'last_transaction_date': ciso8601.parse_datetime(
-                            loan_data['_created']).date()
-                    }
-                    if loan_data.get('state') == LoanState.ITEM_ON_LOAN:
-                        loans[item_pid]['checkout_date'] = ciso8601.\
-                            parse_datetime(loan_data['start_date']).date()
-                        loans[item_pid]['due_date'] = ciso8601.\
-                            parse_datetime(loan_data['end_date']).date()
-                return loans
+            """Generate CSV records."""
+
+            def _process_item_types_libs_locs(csv_data):
+                """Process item types, library and locations data."""
+                # item_type
+                csv_data['item_type'] = self.get_resource(
+                    ItemType, csv_data.get('item_type_pid')).get('name')
+                # temporary item_type
+                if temporary_item_type_pid := csv_data.pop(
+                        'temporary_item_type_pid', None):
+                    csv_data['temporary_item_type'] = self.get_resource(
+                        ItemType, temporary_item_type_pid).get('name')
+                # library
+                csv_data['item_library_name'] = self.get_resource(
+                    Library, csv_data.pop('item_library_pid')).get('name')
+                # location
+                csv_data['item_location_name'] = self.get_resource(
+                    Location, csv_data.pop('item_location_pid')).get('name')
```

Comment:
At this time, using resource cache isn't a good idea.
I think it's better to load all resources of a type in one call.
This method must be refactorized when it is possible to load all resources in the cached class.
At least, could you add a "TODO" in code?